RHDEVDOCS-6023: Installing Argo Rollouts CLI (#76625)
openshift/openshift-docs: modules/gitops-argo-rolouts-cli-ovr.adoc

// Module included in the following assemblies:
//
// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
:_mod-docs-content-type: CONCEPT
[id="gitops-argo-rollouts-cli-overview_{context}"]
= Argo Rollouts CLI Overview

> The Argo Rollouts CLI plugin is not required to use Argo Rollouts resources, because you can instead deploy and monitor a Rollout resource using the {OCP} Console or the `oc` command. However, usage of the Argo Rollouts CLI plugin allows you to handle Argo Rollouts resources and display it directly from the command line interface.
> 

Reviewer: This does not seem to be too user-friendly to include why the user does not need it as the first information. This can be restructured so that we give benefits first, for example:

```suggestion
Argo Rollouts CLI is an optional plugin that allows you to handle and monitor Argo Rollouts resources directly through the command line interface instead of using {OCP} web console or the `oc` command. 
```
Also please keep in mind how to refer to objects: 
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

Author: Fixed.

> Using the Argo Rollouts CLI plugin is optional because you can deploy and monitor an application by installing Argo Rollouts using the {gitops-title} Operator. Following the installation of Argo Rollouts, users have the choice to utilize the Argo Rollouts CLI plugin. This plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.

Reviewer 2: >  Using the Argo Rollouts CLI plugin is optional because you can deploy and monitor an application by installing Argo Rollouts using the {gitops-title} Operator. Following the installation of Argo Rollouts, users have the choice to utilize the Argo Rollouts CLI plugin. This plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.

What about:

The Argo Rollouts CLI plugin is not required to use Argo Rollouts, because you can instead deploy and monitor a Rollout resource using just the OpenShift Console or oc command. However, this plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.

(The original text made it sound like the CLI plugin was involved in installing/monitoring the Argo Rollouts installation itself, which it is not.)

Author: Fixed.

> The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, and terminate a failed Argo Rollouts deployment.

Reviewer 2: > The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, and terminate a failed Argo Rollouts deployment.

I would add this to the end:

The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, ~and~ terminate a failed Argo Rollouts deployment, _**and more**_.
(addition is _**bolded**_, and likewise see the ~strikethrough~)

(This makes it clear that the CLI has many many commands/use cases, and it is not just limited to those explicitly listed.)

Author: Fixed.

> The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, terminate a failed Argo Rollouts deployment, and so on.
> 
> The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.

Reviewer: I do not find the first sentence useful to the user. It is better to include exactly what the user can do, instead of how the plugin operates (it is something that is typically done by engineers, but it is unnecessarry in minimalistic user-centric docs)

Also, this would look better if it was a list:
```suggestion
The Argo Rollouts CLI plugin enables you to perform many actions, including the following actions:

* Make changes to an Argo Rollouts image.
* Monitor the progress of an Argo Rollouts promotion.
* Proceed with the promotion steps in a canary deployment.
* Terminate a failed Argo Rollouts deployment.

The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.
```

Author: Fixed.

> The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.

Reviewer: `oc/kubectl`: Is this true for downstream as well? Or is it only `oc` commands for downstream?

Author: Let me check with the SME.

Author: I discussed it a bit more with @eromanova97 and I am keeping the term as "oc kubetcl".
